### ISSUE TITLE
Deserialization error when getSettings on a not-yet existent index

### ISSUE THREAD

**Issue by jbau** [algolia/algoliasearch-client-java#384]
The response from a `getSettings` call on a non-existent index is
```
{"message":"Index does not exist","status":404}
```
`AlgoliaHttpClient` near line 129 tries to deserialize this as an `AlgoliaError`, which doesn't have a `status` attribute, resulting in a Jackson exception

**Comment by ElPicador** [comment]
I tried to call getSettings on a non existing index, but it doesn't throw any exceptions.
```
Index<?> index = client.initIndex("really-non-existing-index");
index.getSettings();
```

Could you provide me a snippet with the error you have?

**Comment by jbau** [comment]
it turns out that we were providing the api client an `ObjectMapper` using Guice that does not ignore unknown properties, so that was causing failures all over the place with various different api calls.  So it's not the api library's issue.

Just to clarify though, for my sanity, we hadn't changed this provided `ObjectMapper` during upgrade from 2.7 to 2.13.  So I guess my question is if 2.13 is using a different set of backend APIs that are providing these unknown attributes in their responses?  Or did the backend APIs recently change?

**Comment by ElPicador** [comment]
There was no change in the backend APIs.
As far as I know, and from `git blame` there was no change in the Jackson annotation or to the `AlgoliaException` class.

On the other end, the deserialization of Algolia objects (like `AlgoliaException`) shouldn't rely on the object mapper you provide.

So there is probably some jackson configuration missing somewhere.

Could you provide me with the configuration of your ObjectMapper so I can reproduce and fix this issue?

**Comment by ElPicador** [comment]
Do you still face the issue?

**Comment by jbau** [comment]
Sorry, forgot to get back to you.  Providing the new `ObjectMapper` did fix this for us.  Here was the delta:

```
   @Named("algolia")
   public ObjectMapper getObjectMapper() {
     ObjectMapper om = new ObjectMapper();
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     return om;
}
```
This is used here:
```
public class AlgoliaAPIClientProvider implements Provider<APIClient> {

  private final ObjectMapper objectMapper;
  private final AlgoliaConfig algoliaConfig;

  @Inject
  public AlgoliaAPIClientProvider(@Named("algolia") ObjectMapper objectMapper, AlgoliaConfig algoliaConfig) {
    this.objectMapper = objectMapper;
    this.algoliaConfig = algoliaConfig;
  }

  @Override
  public APIClient get() {
    return new ApacheAPIClientBuilder(algoliaConfig.getApplicationId(), algoliaConfig.getApiKey()).
        setObjectMapper(objectMapper).build();
  }
}
```

**Comment by ElPicador** [comment]
Indeed there was some missing annotations on the POJOs.

I've fixed it in #387.
